### PR TITLE
feat: use "lit" instead of "lit-html" and "lit-element"

### PR DESCRIPTION
**Pull request by Westbrook**
## Description
Of the below tests (47), 42 show a clear timing preference for the new `lit` versions vs the currently published versions.

In ALL cases the byte count of the two versions prefers the new `lit` versions by about 113KB, a couple (which include as yet unreleased changes to `sp-action-button` show a very large change).

~One unfortunate mitigation was required to work around the issue in https://github.com/lit/lit/issues/1789, for which a PR has already been approved: https://github.com/lit/lit/pull/1790~

We also needed to internalize the `spread` helper to support its ongoing use in some of our tests. With the creation of `@lit-labs`, it may be that a new community supported version of that functionality surfaces soon, but being we only leverage this in tests at current it seemed that bringing it internal to our repo made sense for now.

Assuming the one issue noted above is included in the next RC, I'd like to suggest that we discuss with seriousness moving to `lit` pre-2.0 in our library. In support of that, I could see this change being moved to a `minor` change from its current `patch` level, however I was quite pleased by the lack of changes needed by our library (outside of the `streamingListener` refactor).

## Motivation and Context
Smaller. Faster. Easier to update. New features. 

## How Has This Been Tested?
The full suite of tests, etc.

Perf comments: https://github.com/adobe/spectrum-web-components/pull/1401#issuecomment-885036800
Preview comments: https://github.com/adobe/spectrum-web-components/pull/1401#issuecomment-889079203

## Types of changes
- [x] Dependency update.

## Checklist:
- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by adixon-adobe** [comment]
This looks awesome!!!

I think it'd make sense to do a canary release for this, but I don't see an issue updating before Lit2 is fully approved if our tests are passing. My team can do some testing around the components we use with a canary release.

**Comment by dsadhanala** [comment]
@Westbrook thanks for sharing the pref stats, looks promising! wondering should we bump up the major version to avoid consuming product teams run into any unexpected behavior. if this is something already plan, please ignore my comment 😃

**Comment by Westbrook** [comment]
@dsadhanala this is a great question, and one we do need to put some thought into. As there isn't much work remaining in https://github.com/adobe/spectrum-web-components/issues/865 and https://github.com/adobe/spectrum-web-components/projects/1 it may truly be time to move towards a "major" release. However, it's important to remember that pre-1.0 ALL "minor" releases are treated as breaking changed by package managers in a way that would allow us to make a change like this with the sort of version security that you are asking about without needing to burn the 1.0 release, just yet. Not that we shouldn't go to 1.0, if we're ready. I'd love to have that conversation more in depth there as well!

File: documentation/src/components/adobe-logo.ts
Comment (Westbrook) [reply documentation/src/components/adobe-logo.ts:18]: Originally we moved to this pattern for two reason:
1. Having a central dependency that re-exported the `lit-*` API allowed for us to track the versions thereof in a single place in our repo. Generally, the libraries this track are fairly slow moving, so maybe the benefit here is more ergonomic than practical.
2. Before the `lit@2.0` release, `lit-html`, and the tooling around it, relied heavily on `instanceof` checks that required a shared version across a project. Leveraging a central dependency reduced the changed of package managers accidentally allowing multiple non-matched instances into a project.

I'd say these realities certainly put moving away from this pattern into conversation. I'm not sure that I'd say that reexporting these APIs locks our elements to them in any function manner, however. It would certainly be a breaking change to _stop_ exporting them, but I do see how choosing to basing our elements on another library, as long as the public API to those element remained the same, would be in any way consequential to our consumers. This is one of the reason that, no matter how powerful the pattern may be, we've avoided passing templates into our elements to date. 

Definitely want to discuss this more, so we can get structure a high quality API around this.

**Comment by Westbrook** [comment]
@adixon-adobe I agree about a canary release. We currently have six PRs that I'd like to include in our upcoming regular release:
- https://github.com/adobe/spectrum-web-components/pull/1379
- https://github.com/adobe/spectrum-web-components/pull/1384
- https://github.com/adobe/spectrum-web-components/pull/1387
- https://github.com/adobe/spectrum-web-components/pull/1390
- https://github.com/adobe/spectrum-web-components/pull/1393
- https://github.com/adobe/spectrum-web-components/pull/1400

With the possibility of a few more opening today. I'd like to propose we take out first canary release once those standard releases are out so we can make a clean comparison on just the `lit` changes herein.

**Comment by Westbrook** [comment]
FYI: we've made a canary release of these packages to NPM. So anyone who is interested or able to check these our in advance of full release, they are available as follows:

```
Successfully published:
 - @spectrum-web-components/accordion@0.4.8-alpha.0+f76f322c5
 - @spectrum-web-components/action-bar@0.2.8-alpha.0+f76f322c5
 - @spectrum-web-components/action-button@0.4.8-alpha.0+f76f322c5
 - @spectrum-web-components/action-group@0.4.8-alpha.0+f76f322c5
 - @spectrum-web-components/action-menu@0.9.8-alpha.0+f76f322c5
 - @spectrum-web-components/asset@0.4.6-alpha.0+f76f322c5
 - @spectrum-web-components/avatar@0.7.6-alpha.0+f76f322c5
 - @spectrum-web-components/banner@0.6.4-alpha.74+f76f322c5
 - @spectrum-web-components/base@0.4.4-alpha.74+f76f322c5
 - @spectrum-web-components/bundle@0.20.3-alpha.0+f76f322c5
 - @spectrum-web-components/button-group@0.6.7-alpha.0+f76f322c5
 - @spectrum-web-components/button@0.13.8-alpha.0+f76f322c5
 - @spectrum-web-components/card@0.8.8-alpha.0+f76f322c5
 - @spectrum-web-components/checkbox@0.9.7-alpha.0+f76f322c5
 - @spectrum-web-components/coachmark@0.6.6-alpha.0+f76f322c5
 - @spectrum-web-components/color-area@0.1.5-alpha.0+f76f322c5
 - @spectrum-web-components/color-handle@0.1.5-alpha.0+f76f322c5
 - @spectrum-web-components/color-loupe@0.1.5-alpha.0+f76f322c5
 - @spectrum-web-components/color-slider@0.1.5-alpha.0+f76f322c5
 - @spectrum-web-components/color-wheel@0.1.5-alpha.0+f76f322c5
 - @spectrum-web-components/dialog@0.6.8-alpha.0+f76f322c5
 - @spectrum-web-components/divider@0.2.6-alpha.0+f76f322c5
 - @spectrum-web-components/dropzone@0.7.6-alpha.0+f76f322c5
 - @spectrum-web-components/field-group@0.3.6-alpha.0+f76f322c5
 - @spectrum-web-components/field-label@0.4.6-alpha.0+f76f322c5
 - @spectrum-web-components/icon@0.9.7-alpha.0+f76f322c5
 - @spectrum-web-components/icons-ui@0.6.7-alpha.0+f76f322c5
 - @spectrum-web-components/icons-workflow@0.6.7-alpha.0+f76f322c5
 - @spectrum-web-components/icons@0.7.6-alpha.74+f76f322c5
 - @spectrum-web-components/iconset@0.5.6-alpha.74+f76f322c5
 - @spectrum-web-components/illustrated-message@0.6.7-alpha.0+f76f322c5
 - @spectrum-web-components/link@0.8.6-alpha.0+f76f322c5
 - @spectrum-web-components/menu@0.8.6-alpha.0+f76f322c5
 - @spectrum-web-components/meter@0.3.8-alpha.0+f76f322c5
 - @spectrum-web-components/modal@0.3.6-alpha.0+f76f322c5
 - @spectrum-web-components/overlay@0.11.6-alpha.0+f76f322c5
 - @spectrum-web-components/picker@0.4.6-alpha.0+f76f322c5
 - @spectrum-web-components/popover@0.9.8-alpha.0+f76f322c5
 - @spectrum-web-components/progress-bar@0.2.8-alpha.0+f76f322c5
 - @spectrum-web-components/progress-circle@0.2.6-alpha.0+f76f322c5
 - @spectrum-web-components/quick-actions@0.4.6-alpha.0+f76f322c5
 - @spectrum-web-components/radio@0.7.6-alpha.0+f76f322c5
 - @spectrum-web-components/search@0.8.8-alpha.0+f76f322c5
 - @spectrum-web-components/shared@0.12.4-alpha.0+f76f322c5
 - @spectrum-web-components/sidenav@0.10.1-alpha.0+f76f322c5
 - @spectrum-web-components/slider@0.9.6-alpha.0+f76f322c5
 - @spectrum-web-components/split-button@0.4.8-alpha.0+f76f322c5
 - @spectrum-web-components/split-view@0.2.6-alpha.0+f76f322c5
 - @spectrum-web-components/status-light@0.7.6-alpha.0+f76f322c5
 - @spectrum-web-components/styles@0.9.7-alpha.0+f76f322c5
 - @spectrum-web-components/switch@0.7.7-alpha.0+f76f322c5
 - @spectrum-web-components/tabs@0.5.7-alpha.0+f76f322c5
 - @spectrum-web-components/tags@0.6.8-alpha.0+f76f322c5
 - @spectrum-web-components/textfield@0.8.7-alpha.0+f76f322c5
 - @spectrum-web-components/theme@0.8.8-alpha.0+f76f322c5
 - @spectrum-web-components/thumbnail@0.1.6-alpha.0+f76f322c5
 - @spectrum-web-components/toast@0.8.8-alpha.0+f76f322c5
 - @spectrum-web-components/tooltip@0.8.8-alpha.0+f76f322c5
 - @spectrum-web-components/top-nav@0.2.7-alpha.0+f76f322c5
 - @spectrum-web-components/underlay@0.6.6-alpha.0+f76f322c5
 - @spectrum-web-components/story-decorator@0.3.8-alpha.0+f76f322c5
 - @spectrum-web-components/vrt-compare@0.0.6-alpha.0+f76f322c5
```

I've also updated the commit to `feat` which would translate to a "breaking" pre-1.0 release on our end. 

Looking forward to getting deeper into this with the team!

File: package.json
Comment (joekukish) [inline package.json:217]: @Westbrook since you removed the lit-element and lit-html dependencies above, is this still required?

File: packages/overlay/src/ActiveOverlay.ts
Comment (joekukish) [inline packages/overlay/src/ActiveOverlay.ts:494]: @Westbrook All these signature changes for `_getUpdateComplete` would be a breaking change.

File: packages/overlay/stories/overlay-story-components.ts
Comment (joekukish) [inline packages/overlay/stories/overlay-story-components.ts:35]: @Westbrook same for this, this is technically a breaking change as well.

File: packages/overlay/src/ActiveOverlay.ts
Comment (Westbrook) [reply packages/overlay/src/ActiveOverlay.ts:494]: Yep, that's why I updated the commit to `feat`. While pre-1.0, semver registers any 0.x.0 change as _breaking_. Our release tooling with Lerna and Conventional Commits, however, register any reference to "breaking" as a x.0.0 change, which I'm not sure the work requires and/or we're fully ready for.

I could certainly see discussing thing accounting for our 1.0 release, and if so, I'd love your thoughts on the benefits you see there and any other breaking/major changes you don't see in #865 or https://github.com/adobe/spectrum-web-components/projects/1 that you think we should account for in doing so.

File: test/lit-helpers.ts
Comment (hunterloftis) [inline test/lit-helpers.ts:20]:
```suggestion
/**
 * TODO: replace this with a built-in directive once one is released
 * (https://github.com/lit/lit/pull/1960)
 *
 * Usage:
```

File: packages/textfield/test/textfield.test.ts
Comment (Westbrook) [reply packages/textfield/test/textfield.test.ts]: Returned.

**Review by hunterloftis — DISMISSED** [review]

**Pull request merged**
